protocols/kad/src/behaviour.rs

```diff
@@ -1029,6 +1030,45 @@ where
             );
     }
 
+    fn determine_mode(&mut self) {
```

Comment:
```suggestion
    fn determine_mode_from_external_addresses(&mut self) {
```

Too long? What do you think?